language-plutus-core/examples/Language/PlutusCore/Examples/Data/InterList.hs

```diff
@@ -97,7 +99,7 @@ interCons = runQuote $ do
           , Var () xs
           ]
 
-foldrInterList :: Term TyName Name uni ()
+foldrInterList :: uni `Includes` () => Term TyName Name uni ()
```

Comment:
I generally dislike the idea if having two different `unit`s. What about 

```haskell
-- | @if_then_else_@ as a PLC term.
--
-- > /\(A :: *) -> \(b : Bool) (x y : () -> A) -> IfThenElse {() -> A} b x y ()
ifThenElse :: (TermLike term TyName Name uni, uni `IncludesAll` '[Bool, ()]) => term ()
```

should it also use Scott-encoded `unit`? What if we add a builtin like

```
nil : forall a. () -> list a
```

should we be able to fit it into `ifThenElse` or do we need to provide the `scottToMetaUnit` and `metaToScottUnit` adaptors?

Rather than choosing between different `unit`s on a case-by-case basis I'd rather use the same `unit` everywhere, it's not a big deal to add it to the universe and the user still has to deal with some particular universe.

Reply:
I don't mind hugely for the tests. I had in mind that it would give us a clear policy:
- Use builtin types iff you are interacting with a builtin function that uses them.
- Use compiled/Scott-encoded types in all other cases.

I guess for the tests it also makes them more readable.

Reply:
We could add `unit` right into the AST by having something like

```haskell
data OrUnit a = Unit | NotUnit a

data Type tyname uni ann
    = <...>
    | TyBuiltin ann (OrUnit (Some (TypeIn uni)))

data Term tyname name uni ann
    = <...>
    | Constant ann (OrUnit (Some (ValueOf uni)))
```

if it's such a special type, but I'd just prefer to use ``uni `Includes` ()` everywhere.

Anyway, I'm going to finish this off ASAP and move to the other project and then feel free to change stuff the way you like it.